src/OpenTK/Toolkit.cs

```diff
@@ -152,10 +152,27 @@ public static Toolkit Init(ToolkitOptions options)
                          * NOTE:
                          * Non-Windows platforms should be handled via the OpenTK.dll.config file as appropriate
                          */
-                        string path = Path.GetDirectoryName(Assembly.GetEntryAssembly().Location);
-                        path = Path.Combine(path, IntPtr.Size == 4 ? "x86" : "x64");
-                        bool ok = SetDllDirectory(path);
-                        if (!ok) throw new System.ComponentModel.Win32Exception();
+                        Assembly entryAssembly = Assembly.GetEntryAssembly();
+                        if (entryAssembly != null)
+                        {
+                            try
+                            {
+                                string assemblyLocation = entryAssembly.Location;
+                                string path = Path.GetDirectoryName(assemblyLocation);
+                                path = Path.Combine(path, IntPtr.Size == 4 ? "x86" : "x64");
+                                bool ok = SetDllDirectory(path);
+                                if (!ok)
+                                {
+                                    //A fairly fundamental Win32 syscall failed. Developer probably wants to know about this, but not necessarily users
+                                    throw new System.ComponentModel.Win32Exception("Windows reported that attempting to set the DLL search path failed");
+                                }
+                            }
+                            catch
+                            {
+                                //A fairly fundamental Win32 syscall failed. Developer probably wants to know about this, but not necessarily users
+                                throw new System.ComponentModel.Win32Exception("An unexpected Win32 error occured when attempting to set the DLL search path");
+                            }
```

Comment:
It's unclear why you are throwing an exception, just to rethrow it with a different error message. It will also remove the `GetLastError` that was present in the first exception and make that information not available.

Reply:
Good point actually......
The inital code had the #ifdef in (and I suppose could have rethrown the exception in question, rather than trying to generate a new one), the #ifdef basically just got removed. I also noticed that the Path call could conceivably throw an exception, so dropped that into the catch block too.....

Reply:
Yours is a good idea, although at that point, I might add back in rethrow under an #ifdef
That would then give an error message, with instructions to substitue a debug build and retry for interested developers?